diffuse-sharpen: adjust labels and tooltips

quovadit opened the pull request:
In order to make the great tool 'diffuse or sharpen' even more usable, I would suggest to adapt a few labels and tooltips..

My main motivation was the fact, that some users will use it just for sharpening, and the only slider named 'sharpness' is not recommended as a standalone sharpening tool.

That's why I renamed these labels:
- "sharpness" -> "sharpness (highest frequencies)"
- "diffusion speed" -> "speed (sharpen ↔ diffuse)"

I also moved tooltips, which were the same for all 4 sliders to the heading, and adapted the wording of some tooltips according to dtdocs and aureliens youtube-video.

@aurelienpierre : I hope I did not mess it up completely...

elstoc commented on src/iop/diffuse.c:
```diff
@@ -1514,98 +1515,85 @@ void gui_init(struct dt_iop_module_t *self)
-                                          "negative values sharpen.\n"
-                                          "zero does nothing."));
+  gtk_widget_set_tooltip_text(g->first, _("speed of low-frequency wavelet layers\n"
+                  "in the direction of 1st order anisotropy (set below)."));
```
I understand you're trying to remove repetition here but honestly it might be better to repeat than to make people move around to view another tooltip.

quovadit replied:
It's not just about removing repetition, but making it easier to understand the tooltips.
If I want to know, what's the difference between the 4 sliders, I have to filter out everything which is identical, in order to see the differences.
Currently the start and the end of the tooltips are identical for all 4 sliders, but there are small differences inbetween.

As an alternative I now start with the differences, and after an empty line add the repetition.

elstoc reviewed (commented):
A few comments...

quovadit commented on src/iop/diffuse.c:
```diff
@@ -1491,20 +1491,21 @@ void gui_init(struct dt_iop_module_t *self)
   dt_bauhaus_slider_set_format(g->radius_center, "%.0f px");
   gtk_widget_set_tooltip_text(
-      g->radius_center, _("main scale of the diffusion.\n"
+      g->radius_center, _("main scale of the diffusion in px.\n"
```
my fault, did not see that, reverted

quovadit commented on src/iop/diffuse.c:
```diff
@@ -1514,98 +1515,85 @@ void gui_init(struct dt_iop_module_t *self)
   dt_bauhaus_slider_set_format(g->radius, "%.0f px");
   gtk_widget_set_tooltip_text(
-      g->radius, _("width of the diffusion around the center radius.\n"
+      g->radius, _("width of the diffusion around the central radius in px.\n"
```
same

quovadit commented on src/iop/diffuse.c:
```diff
@@ -54,7 +54,7 @@ typedef struct dt_iop_diffuse_params_t
   // global parameters
   int iterations;           // $MIN: 1   $MAX: 128   $DEFAULT: 1  $DESCRIPTION: "iterations"
-  float sharpness;          // $MIN: -1.  $MAX: 1.   $DEFAULT: 0. $DESCRIPTION: "sharpness"
+  float sharpness;          // $MIN: -1.  $MAX: 1.   $DEFAULT: 0. $DESCRIPTION: "sharpness (highest frequencies)"
```
> unless there was another sharpness parameter

That's exactly the reason why I changed this and the next one.
It is a module to diffuse and to sharpen, and there is only one slider named 'sharpness'.
This implies, that this is the way you do sharpening with this module.
But in fact, Aurelien says in his video, it is not suggested to use this slider as a standalone sharpening tool.

quovadit commented on src/iop/diffuse.c:
```diff
@@ -1514,98 +1515,85 @@ void gui_init(struct dt_iop_module_t *self)
+                   "if you plan on deblurring, \n"
+                   "the radius should be around the width of your lens blur."));
+
+  GtkWidget *label_speed = dt_ui_section_label_new(_("speed (sharpen ↔ diffuse)"));
```
As explained above, I think this is the most important change of this PR.
Every time I use the module, I want to instantly understand, which sliders to use in which direction in order to sharpen / diffuse.
So this information can't be (only) in a tooltip (no matter where exactly)

quovadit commented on src/iop/diffuse.c:
```diff
@@ -1514,98 +1515,85 @@ void gui_init(struct dt_iop_module_t *self)
-                                "negatives make the diffusion follow gradients more closely"));
+  gtk_widget_set_tooltip_text(g->anisotropy_fourth,_("direction of 4th order speed (set above)\n"));
 
   gtk_box_pack_start(GTK_BOX(self->widget), dt_ui_section_label_new(_("edges management")), FALSE, FALSE, 0);
```
done

aurelienpierre commented:
> "sharpness" -> "sharpness (highest frequencies)"

The sharpness boosts the wavelets coeffs for all the frequencies selected. They are "high" frequencies only in the sense that each wavelet scale gets a "high frequency" layer (the wavelets coeffs/details/oscillations around the local average) and a "low frequency" layer (the local average), that will be split again at the next scale while the final scale (the residual) is not reached. Where it becomes nasty is that the wavelet scales are analogous to frequencies in a Fourier (spectral) space.

I'm not sure what's best, and again, trying to be "general audience" in this context only increases the confusion, I guess. But "highest frequencies" kinda implies high frequencies as per Fourier spectral formalism.

> "diffusion speed" -> "speed (sharpen left_right_arrow diffuse)"

Again, not sure. Speed of what ?  That's unclear now.

quovadit commented:
> "sharpness" -> "sharpness (highest frequencies)"

I just took the existing tooltip and moved it to the label. 
Of course you know exactly which wording is correct.
My intention was just to say: 'this is not the standalone sharpness slider for you'

> Speed of what ? That's unclear now.

I removed 'diffusion' from 'diffusion speed' because the module is about diffusion AND sharpening.
So 'diffusion speed' seems incorrect for me, if I use negative values.

quovadit commented:
> negative diffusion is what it's doing

sure, that's technically correct, but I still think it's very important here to have 'sharpen <-> diffuse' in the label.
And 'diffusion speed (sharpen <-> diffuse)' is quite long...

The slider also just says '1st order speed', and not '1st order diffusion speed'

So is it enough to add 'diffusion speed' in the tooltip?
('speed of low-frequency wavelet layers ...')

quovadit commented:
@TurboGit : as 'end-November' is over: will this still be in 3.8

TurboGit commented:
> @TurboGit : as 'end-November' is over: will this still be in 3.8

No, a bit too late for 3.8 and I feel the discussion is not over yet. I'll re-target for 4.0.

aurelienpierre reviewed (approved):
I'm ok with that. @quovadit could you fix conflicts ?

TurboGit reviewed (commented):
Thanks!

kofa73 commented:
The file was merged with unresolved conflicts!
https://github.com/darktable-org/darktable/commit/e363e46a2a8d4106822030ef499e4cd96be4f048#diff-e7424db92a0af5cdc106e18457769e75e88c3b70bbac59b7bf52568691f8afc0R1509

TurboGit commented:
How was this possible? The button was green, meaning no conflict!

The only explanation is that the files were updated with conflicts in them and force pushed here. Is that the case @quovadit?

It was approved, so I didn't looked at the latest version, that's in any case my fault at then end.

TurboGit commented:
As an immediate action I have reverted the 3 commits. Please @quovadit provide a new clean PR for this. Thanks.

quovadit commented:
I rebased and did not (!) get any conflicts
$ git fetch upstream
$ git rebase -i upstream/master

... so I force-pushed

thanks for reverting, sorry for the confusion, I will provide a new clean PR
